File: test-apps/scripting/src/main/resources/ruby/simple-batchlet.rb
=begin
 Copyright (c) 2014 Red Hat, Inc. and/or its affiliates.

 This program and the accompanying materials are made
 available under the terms of the Eclipse Public License 2.0
 which is available at https://www.eclipse.org/legal/epl-2.0/

 SPDX-License-Identifier: EPL-2.0
=end

def stop()
    puts 'In stop function'
end

# access built-in variables: jobContext, stepContext and batchProperties,
# set job exit status to the value of testName property, and
# return the value of testName property as step exit status,
#
def process()
    puts('jobName: ' + $jobContext.getJobName())
    puts('stepName: ' + $stepContext.getStepName())
    testName = $batchProperties.get('testName')
    $jobContext.setExitStatus(testName)
    return testName
end

File: test-apps/scripting/src/main/resources/php/simple-batchlet.php
/*
 * Copyright (c) 2014 Red Hat, Inc. and/or its affiliates.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 * Cheng Fang - Initial API and implementation
 */

<?php
    import java.util.HashMap;

    echo "jobName: " . $jobContext->getJobName() . PHP_EOL;
    echo "stepName: " . $stepContext->getStepName() . PHP_EOL;

    $testName = $batchProperties["testName"];
    $jobContext->setExitStatus($testName);
    return $testName;
?>